vspace/src/main/java/edu/asu/diging/vspace/core/services/impl/ImageService.java

```diff
@@ -82,92 +84,120 @@ public ImageData getImageDimensions(IVSImage image, int width, int height) {
         Float newHeight = (new Float(width) / new Float(imageWidth)) * new Float(imageHeight);
         return new ImageData(newHeight.intValue(), width);
     }
-    
+
     /**
-     * Method to return the requested images  
+     * Method to return the requested images
      * 
-     * @param pageNo. if pageNo<1, 1st page is returned, if pageNo>total pages,last page is returned
+     * @param pageNo. if pageNo<1, 1st page is returned, if pageNo>total pages,last
+     *                page is returned
      * @return list of images in the requested pageNo
-     */ 
+     */
     @Override
-    public List<IVSImage > getImages(int pageNo) {
+    public List<IVSImage> getImages(int pageNo) {
         pageNo = validatePageNumber(pageNo);
-        Pageable sortByRequestedField = PageRequest.of(pageNo-1, pageSize, Sort.by(SortByField.CREATION_DATE.getValue()).descending());
+        Pageable sortByRequestedField = PageRequest.of(pageNo - 1, pageSize,
+                Sort.by(SortByField.CREATION_DATE.getValue()).descending());
+        Page<VSImage> images = imageRepo.findAll(sortByRequestedField);
+        List<IVSImage> results = new ArrayList<>();
+        images.getContent().forEach(i -> results.add(i));
+        return results;
+    }
+
+    /**
+     * Method to return the requested images
+     * 
+     * @param pageNo. if pageNo<1, 1st page is returned, if pageNo>total pages,last
+     *                page is returned
+     * @return list of images in the requested pageNo and requested order.
+     */
+    @Override
+    public List<IVSImage> getImages(int pageNo, String sortedBy, String order) {
+        Sort sortingParameters = Sort.by(SortByField.CREATION_DATE.getValue()).descending();
+        pageNo = validatePageNumber(pageNo);
+        if (sortedBy != null && !sortedBy.equals("") && SortByField.getAllValues().contains(sortedBy)) {
+            if (order.equalsIgnoreCase(Sort.Direction.DESC.toString())) {
```

Comment:
if only order is given but not sortedBy, we probably still want to use follow the order. Right now it is only used when sortBy is provided.